CodeChallenge/Repositories/EmployeeRespository.cs

```diff
@@ -32,6 +32,24 @@ public Employee GetById(string id)
             return _employeeContext.Employees.SingleOrDefault(e => e.EmployeeId == id);
         }
 
+        public Employee GetEmplyeeAndReportsById(String id)
+        {
+            var employee = _employeeContext.Employees.Where(e => e.EmployeeId == id)
+                                                    .Include(e => e.DirectReports)
+                                                    .SingleOrDefault();
+            if (employee.DirectReports != null)
+            {
+                // TODO: This is terribly ineffcient, however, I could not find a 
+                // better way to load the data recursively without just loading the
+                // whole table. I would love to know the 'proper' way to do what I am trying.
+                for (int i = 0; i < employee.DirectReports.Count; i++)
+                {
+                    employee.DirectReports[i] = GetEmplyeeAndReportsById(employee.DirectReports[i].EmployeeId);
```

Comment:
I would love to know the 'proper' way to load from the db

Reply:
Upon further testing, there seems to be more wrong with this method of loading. When running all tests at once, 'John Lennon' will only load 'Paul McCartney' as a direct report, but running the tests one at a time works fine.